frontend/src/components/ui/pagination.ts

```diff
@@ -120,9 +139,17 @@ export class Pagination extends LitElement {
     `,
   ];
 
-  @property({ type: Number })
+  searchParams = new SearchParamsController(this, (params) => {
+    const page = parsePage(params.get(this.name));
+    this.onPageChange(page);
```

Comment:
This seems to break going back/forwards since `onPageChange` calls `searchParams.set`. I'm able to make it work by replacing this line with:

```ts
if (this.page !== page) {
  this.dispatchEvent(
    new CustomEvent<PageChangeDetail>("page-change", {
      detail: { page: page, pages: this.pages },
      composed: true,
    }),
  );
  this.page = page;
}
```

Or you could modify `onPageChange` to skip `this.searchParams.set` on popstate.

Reply:
Good catch, thank you!